compiler-plugin/src/test/kotlin/com/google/devtools/ksp/processor/AnnotatedUtilProcessor.kt

```diff
@@ -0,0 +1,129 @@
+import com.google.devtools.ksp.visitor.KSTopDownVisitor
+import kotlin.jvm.Throws
+import kotlin.reflect.KClass
+
+class AnnotatedUtilProcessor : AbstractTestProcessor() {
+    val results = mutableListOf<String>()
+    private val visitors = listOf(GetAnnotationVisitor(), GetAnnotationsVisitor(), GetAnnotationsByTypeVisitor())
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        resolver.getSymbolsWithAnnotation("Test", true).forEach {
+            val annotation = it.annotations.single()
+            annotation.arguments.forEach { visitors.forEach { visitor -> it.accept(visitor, results) } }
+        }
+        return emptyList()
+    }
+
+    override fun toResult(): List<String> {
+        return results
+    }
+
+    inner class GetAnnotationVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            if (annotated.isAnnotationPresent(ParametersTestAnnotation::class)) {
+                annotated.getAnnotation(ParametersTestAnnotation::class)?.let { data.add(it.toString()) }
+            }
+            if (annotated.isAnnotationPresent(ParameterArraysTestAnnotation::class)) {
+                annotated.getAnnotation(ParameterArraysTestAnnotation::class)?.let { data.add(it.toString()) }
+            }
+            if (annotated.isAnnotationPresent(ParametersTestWithNegativeDefaultsAnnotation::class)) {
+                annotated.getAnnotation(ParametersTestWithNegativeDefaultsAnnotation::class)
+                    ?.let { data.add(it.toString()) }
+            }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+
+    inner class GetAnnotationsVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            annotated.getAnnotations().forEach { data.add(it.toString()) }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+
+    inner class GetAnnotationsByTypeVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            annotated.getAnnotationsByType(ParametersTestAnnotation::class).let { data.add(it.toString()) }
+            annotated.getAnnotationsByType(ParameterArraysTestAnnotation::class).let { data.add(it.toString()) }
+            annotated.getAnnotationsByType(ParametersTestWithNegativeDefaultsAnnotation::class)
+                .let { data.add(it.toString()) }
+            annotated.getAnnotationsByType(Throws::class).let { data.add(it.toString()) }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+
+    @Suppress("LongParameterList")
+    annotation class ParametersTestAnnotation(
```

Comment:
Does your code support instances of `kotlin.Annotation` subclasses as a type of annotation class values? Kotlin documentation states:
> Allowed parameter types:
> - [...]
> - Other annotations
> - [...]

[Source](https://kotlinlang.org/docs/annotations.html#constructors)

Reply:
No it doesn't, I didn't take that into consideration. I can add that functionality.

Reply:
This is a port from my repo https://github.com/AlexBeggs/ksp-annotation-parser, I am currently implementing tests for this repo, but I ran into some issues over the weekend. Finally got it worked out last night.

Reply:
I really need this functionality. It would be very nice if it would be added to ksp :)